xarray/core/alignment.py

```diff
@@ -145,7 +146,14 @@ def __init__(
         self.objects = tuple(objects)
         self.objects_matching_indexes = ()
 
-        if join not in ["inner", "outer", "override", "exact", "left", "right"]:
+        if not isinstance(join, CombineKwargDefault) and join not in [
+            "inner",
+            "outer",
+            "override",
+            "exact",
+            "left",
+            "right",
+        ]:
```

Comment:
Assuming these strings are part of a string literal type, could [`typing.get_args`](https://docs.python.org/3/library/typing.html#typing.get_args) be useful here?

Reply:
ooh probably! I hadn't seen that before 🚀